### PR TITLE
Chat - Mobile Frontend

### PR DESCRIPTION
**Pull request by petabite**
### Administrative Info
Monday Board ID: 853642866
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
- style ChatScreen according to mockups
- remove Chat tab for guests
- remove rating functionality that was implemented in #60
- BUGFIX: fix messages showing out of order (introduced in #60)

### Testing
- tested sending messages between mobile and the admin portal and ensured the user flow followed the mockup

### Confirmation of Change 
<p float="left">

</p>

In order: 
1. Chat prompt (before any chat room is created)
2. Message bubbles
3. Inputting message (KeyboardAvoidingView)
4. Resolved chat state
5. Need more help pop-up

### CODE REVIEW

File: mobile/src/screens/ChatScreen.tsx [hunk 1]
```diff
@@ -49,122 +58,294 @@ export const ChatScreen: React.FC = () => {
...
-    if (newRoomId && auth.user) {
-      chatAPI.sendMessage(newRoomId, messageText, auth.user?.name);
-      setMessageText("");
+    if (auth.user) {
+      // Create a new room if no previous roomId OR if current chat is resolved
+      if (!newRoomId || (chatRoomData && chatRoomData.resolved)) {
+        setMessages([]);
+        newRoomId = await chatAPI.createRoom(auth.user);
+        setRoomId(newRoomId);
+        AsyncStorage.setItem('chatRoomId', newRoomId);
```
Comment (dacruz21) [inline mobile/src/screens/ChatScreen.tsx:79]: Hmm we might actually need to scope this to the current user, so this doesn't happen:

1. user1 signs in
2. user1 creates chat room
3. user1 signs in, user2 signs out
4. user2 opens chat, cached room ID is not accessible to them

definitely an edge case and I don't think many families will use multiple accounts but it's an easy fix, just append the user ID to the asyncstorage key

File: mobile/src/screens/ChatScreen.tsx [hunk 2]
```diff
@@ -49,122 +58,294 @@ export const ChatScreen: React.FC = () => {
...
+                                      : Colors.shadowColor,
+                                },
+                              ]}
+                            >
+                              {text}
+                            </Text>
+                          </View>
+                        </View>
+                      );
+                  })}
```
Comment (dacruz21) [inline mobile/src/screens/ChatScreen.tsx:145]: Can you clean this up into a `ChatBubble` component please? It's a little hard to follow atm

File: mobile/src/screens/ChatScreen.tsx [hunk 3]
```diff
@@ -49,122 +58,294 @@ export const ChatScreen: React.FC = () => {
...
+          <TouchableOpacity
+            onPress={() => setShowMoreHelp(false)}
+            style={styles.closeButtonContainer}
+          >
+            <Image
+              style={styles.closeButtonIcon}
+              source={require('../../assets/images/times-solid.png')}
+            />
+          </TouchableOpacity>
+        </View>
```
Comment (dacruz21) [inline mobile/src/screens/ChatScreen.tsx:259]: Maybe also make this a component

**Review by typokign — CHANGES_REQUESTED** [review]
Looks close! DMed you a couple things on slack and a few comments below

File: mobile/assets/i18n/en.json [hunk 4]
```diff
@@ -42,5 +40,9 @@
   "forgotPassword": "Forgot Password?",
   "resetPasswordInstructions": "Please enter the email address you signed up with and we'll send you instructions on resetting your password",
   "sendPasswordResetEmail": "Send",
-  "passwordResetEmailSent": "Password reset email sent"
+  "passwordResetEmailSent": "Password reset email sent",
+  "needMoreHelp": "Need more help? Click here.",
```
[the remark this answers is not in the document]
Comment (petabite) [reply mobile/assets/i18n/en.json:44]: Yeah it does. Agreed. I'll change that.

File: mobile/src/screens/ChatScreen.tsx (hunk 1 above)
Comment (petabite) [reply mobile/src/screens/ChatScreen.tsx:79]: Oh yeah, nice catch. That would definitely be an issue!

**Comment by dacruz21** [comment]
Send button is misaligned on my phone but otherwise looks good!

**Review by typokign — DISMISSED** [review]
